File: app/src/main/java/com/gnoemes/shikimoriapp/entity/anime/domain/AnimeType.java
package com.gnoemes.shikimoriapp.entity.anime.domain;

public enum AnimeType {
    TV("tv"),
    MOVIE("movie"),
    OVA("ova"),
    ONA("ona"),
    SPECIAL("special"),
    MUSIC("music"),
    TV_13("tv_13"),
    TV_24("tv_24"),
    TV_48("tv_48");


    private final String type;

    AnimeType(String type) {
        this.type = type;
    }

    public boolean equalsType(String otherType) {
        return type.equals(otherType);
    }

    @Override
    public String toString() {
        return this.type;
    }
}

File: app/src/main/java/com/gnoemes/shikimoriapp/entity/anime/domain/AnimeType.kt
package com.gnoemes.shikimoriapp.entity.anime.domain

import com.google.gson.annotations.SerializedName

enum class AnimeType {
    @SerializedName("tv")
    TV,
    @SerializedName("movie")
    MOVIE,
    @SerializedName("ova")
    OVA,
    @SerializedName("ona")
    ONA,
    @SerializedName("special")
    SPECIAL,
    @SerializedName("music")
    MUSIC,
    @SerializedName("tv_13")
    TV_13,
    @SerializedName("tv_24")
    TV_24,
    @SerializedName("tv_48")
    TV_48,
    @SerializedName("")
    NONE;

    override fun toString(): String {
        return name.toLowerCase()
    }
}